### ISSUE TITLE
[Download] Size of asset is not visible

### ISSUE THREAD

**Issue by bmalecki** [adobe/aem-core-wcm-components#819]
## Feature Request

**Is your feature request related to a problem? Please describe.**
When an uploaded asset does not contain metadata **dam:size** (e.g. `/content/dam/your-app/example-image.jpg/jcr:content/metadata`), size of asset is not visible on Download component.

**Describe the solution you'd like**
To fix this issue we need to change DownloadImpl code in Core Components Repository (https://github.com/adobe/aem-core-wcm-components/blob/master/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/DownloadImpl.java#L209)

from
```
size = null;
Object rawFileSizeObject = downloadAsset.getMetadata(DamConstants.DAM_SIZE);
if (rawFileSizeObject != null) {
long rawFileSize = (Long) rawFileSizeObject;
size = FileUtils.byteCountToDisplaySize(rawFileSize);
}
```
to
```
long rawFileSize = downloadAsset.getOriginal().getSize();
size = FileUtils.byteCountToDisplaySize(rawFileSize);
```

**Are there alternatives?**
Another solution to resolve this issue is run "Add asset size" workflow over an asset. This workflow adds `dam:size` metadata to assets.

**Comment by andreeadracea** [comment]
The problem was solved.

**Comment by richardhand** [comment]
The code for this issue is merged to master and will be available in the upcoming [2.8.0 release](https://github.com/adobe/aem-core-wcm-components/milestone/18).
